aten/src/ATen/native/cuda/Sort.cu

```diff
@@ -46,7 +50,186 @@ bool should_use_th_sort(const Tensor &self, int64_t dim) {
 
 std::vector<int64_t> infer_dense_strides_dim_last(const Tensor & self, int64_t dim);
 
-// We perform a vectorized segmented sort in cub.
+void fillSliceWithIndex(Tensor& t,
+                        int dim) {
+  if (t.numel()) {
+    auto sizes = DimVector(t.dim(), 1);
+    sizes[dim] = t.sizes()[dim];
+    int64_t rangelen = 1;
+    rangelen = std::max(t.sizes()[dim], rangelen);
```

Comment:
nit: if t.nume() is non-zero, t.sizes()[dim] cannot be 0, so you don't need this std::max?

Reply:
I tried that, but it seemed to break with `torch.tensor(1, device='cuda')` in some test cases

Reply:
Well, that's a different problem, you can't call t.sizes()[dim] on a zero-dim tensor - that's an out of bounds access that's bad in any case and should be fixed. I though zero-dim tensors can't come here (you used to turn them into 1d previously).

Reply:
I think the test that does this is [here](https://github.com/pytorch/pytorch/blob/bac4cfd54d44aa0fbc574e6561b878cb406762cc/test/test_sort_and_select.py#L281)

Reply:
I think if you do https://github.com/pytorch/pytorch/pull/55392#discussion_r619534695, the problem will be solved automatically, because for this path t will not be 0D tensors.